src/kernel/include/inv.h

```diff
@@ -90,6 +90,26 @@ sret_deactivate(struct cap_captbl *t, capid_t capin, livenessid_t lid)
 { return cap_capdeactivate(t, capin, CAP_SRET, lid); }
 
 static int
+asnd_construct(struct cap_asnd *asndc, struct cap_arcv *arcvc, capid_t rcv_cap, u32_t budget, u32_t period)
```

Comment:
I still am a bit confused right now. There are 2 scenario here to call `asnd_construct`.
1. For `hw_attach_rcvcap`: In this case we cannot use `asnd_activate` so we call `asnd_construct` directly. I guess this is Okay for now. (Except for synchronization fixes that it would require.)
2. From `asnd_activate` for `cos_asnd_alloc` API: If we could just call `asnd_construct` from here, we could avoid code duplication. Its a big mistake, I should have considered it! Sorry.. And I guess you're referring to this case in your previous comments? 

That was the confusion, all this while I was thinking it was the first case you were referring to and I wasn't sure how to call `asnd_activate` from `hw_attach_rcvcap`. Now if I got it right, your comment isn't for that case, its for 2nd one. And if that's the case, yes, I'll change it and I'll keep in mind for any such things I do in future.